CMakeLists.txt

```diff
@@ -1310,6 +1310,7 @@ add_subdirectory(plugins/network)
 add_subdirectory(plugins/resources)
 add_subdirectory(plugins/rule_engines)
 add_subdirectory(test/c_api_test)
+add_subdirectory(unit_tests)
```

Comment:
Thoughts on putting this under the test directory? Or do we want to phase that out eventually?